modules/tinymce/src/plugins/advlist/main/ts/core/ListUtils.ts

```diff
@@ -1,18 +1,24 @@
-import { Optional, Type } from '@ephox/katamari';
+import { Arr, Optional, Type } from '@ephox/katamari';
 
 import Editor from 'tinymce/core/api/Editor';
 
+const isCustomList = (list: Element): boolean =>
+  /\btox\-/.test(list.className);
+
 const isChildOfBody = (editor: Editor, elm: Node): boolean => {
   return editor.dom.isChildOf(elm, editor.getBody());
 };
 
-const isTableCellNode = (node: Node | null): boolean => {
-  return Type.isNonNullable(node) && /^(TH|TD)$/.test(node.nodeName);
-};
+const matchNodeNames = <T extends Node = Node>(regex: RegExp) =>
+  (node: Node | null): node is T => Type.isNonNullable(node) && regex.test(node.nodeName);
 
-const isListNode = (editor: Editor) => (node: Node | null): boolean => {
-  return Type.isNonNullable(node) && (/^(OL|UL|DL)$/).test(node.nodeName) && isChildOfBody(editor, node);
-};
+const isListNode = matchNodeNames<HTMLOListElement | HTMLUListElement | HTMLDListElement>(/^(OL|UL|DL)$/);
+
+const isTableCellNode = matchNodeNames<HTMLTableHeaderCellElement | HTMLTableCellElement>(/^(TH|TD)$/);
+
+const inList = (editor: Editor, e: Node[], nodeName: string): boolean =>
+  Arr.findUntil(e, isListNode, isTableCellNode)
+    .exists((list) => list.nodeName === nodeName && !isCustomList(list) && isChildOfBody(editor, list));
```

Comment:
Minor: should the `isCustomList` check be moved so that it's earlier? Should it make it fail the `Arr.findUntil` for example?

Reply:
Make sense to move the `isCustomList` earlier. Made the suggested changes, thanks.